beater/beater.go

```diff
@@ -214,6 +216,33 @@ func (bt *beater) start(ctx context.Context, cancelContext context.CancelFunc, b
 		if err := reloader.reload(bt.rawConfig, "default", nil); err != nil {
 			return nil, err
 		}
+		go func() {
+			sigs := make(chan os.Signal, 1)
+			defer close(sigs)
+			signal.Notify(sigs, syscall.SIGUSR1)
+			for {
+				select {
+				case <-sigs:
+					rawcfg, err := common.LoadFile("./apm-server.dev.yml")
+					if err != nil {
+						fmt.Printf("failed to load apm-server.dev.yml! %v\n", err)
+						continue
+					}
+					integrationConfig, err := config.NewIntegrationConfig(rawcfg)
+					if err != nil {
+						fmt.Println("failed to get apm config:", err)
+						continue
+					}
+					apmServerCommonConfig := integrationConfig.APMServer
+					if err := reloader.reload(apmServerCommonConfig, "default", nil); err != nil {
+						fmt.Println("got err while reloading:", err)
+						continue
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
```

Comment:
for local testing

Reply:
Is this something we want or think would be valuable? Happy to make the suggested improvements, just not sure if this is something we want in the project.

Reply:
I'd prefer not to add it right now. If you were planning to remove it before merging then just dismiss my comment